distrho/src/DistrhoPluginVST.cpp

```diff
@@ -1058,7 +1082,41 @@ static intptr_t vst_dispatcherCallback(AEffect* effect, int32_t opcode, int32_t
             return 1;
         }
         return 0;
+    case effGetParameterProperties:
+        if (ptr != nullptr && index < static_cast<int32_t>(plugin.getParameterCount()))
+        {
+            const uint32_t hints = plugin.getParameterHints(index);
+            
+            if (hints & kParameterIsOutput)
+                return 1;
 
+            if (VstParameterProperties* const properties = (VstParameterProperties*)ptr) 
+            {
+                if (hints & kParameterIsInteger)
+                {
+                    properties->flags = kVstParameterUsesIntStep | kVstParameterUsesIntegerMinMax;
+                    const ParameterRanges& ranges(plugin.getParameterRanges(index));
+
+                    properties->stepInteger = 1;
+                    properties->largeStepInteger = 1;
+                    properties->minInteger = static_cast<int32_t>(ranges.min);
+                    properties->maxInteger = static_cast<int32_t>(ranges.max);
+                }
+                else if (hints & kParameterIsBoolean)
```

Comment:
What about parameters that are both an int and a bool? Should I flip these ifs around? Or consider both flags, and print an int instead of a float in ParamDisplay?

Reply:
ah yes, you should check for booleans first, then integers.